### PR TITLE
Use typing namespace

### PR DESCRIPTION
**Pull request by venthur**
this PR replaces `import typing.X; X` with `import typing; typing.X`

it is based on https://github.com/urllib3/urllib3/pull/2779 which is not merged yet.

closes: #2789

### CODE REVIEW

File: changelog/2789.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Changed the code so we use the typing namespace consistently throughout the
```
Comment (sethmlarson) [inline changelog/2789.bugfix.rst:1]: I don't think we need a changelog entry, this change isn't user-facing?

Comment (venthur) [reply changelog/2789.bugfix.rst]: alright, done. :)

**Comment by venthur** [comment]
Is there something that should be merged before this PR can be considered?

**Review by sethmlarson — APPROVED** [review]
Just a lot to read even if it's all cursory glances ;) Let's merge this!

**Comment by venthur** [comment]
Awesome, thanks! 

It's a bit unfortunate that the PR was squashed, because I based my changes on @graingert's [PR](https://github.com/urllib3/urllib3/pull/2779). I hope you don't mind @graingert !

**Comment by sethmlarson** [comment]
@venthur I used the GitHub commit coauthor feature for the commit so you both get credit: https://github.com/urllib3/urllib3/commit/fe39ebfb948db43002547b9258273182705b5434

**Comment by venthur** [comment]
@sethmlarson I didn't notice that, thank you!

**Pull request merged**